### PR TITLE
Fixes#1580 Pagination added in imagesearch

### PR DESCRIPTION
**Pull request by ujjwalagrawal17**
## Fixes#1580 Pagination added in image search.

Fixes#1580 (Search results sometimes appear twice)

## Description (required)
- Added a parameter "sroffset" for getting list of images in search. (It helps in providing images after that offset)
- On last recieved of recyclerview we are calling API again with offset = size of query list it gives us next 25 images .
- We add the images to the already existing list and refresh the adapter.

## Tests performed 
Manually Tested on {API 25 & Moto G5S+}, with ProdDebug variant.

### CODE REVIEW

**Comment by ujjwalagrawal17** [comment]
@nicolas-raoul @neslihanturan can you test this feature. Any suggestions on the implementation would be helpful.

File: app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java [hunk 1]
```diff
@@ -70,6 +70,17 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,Bundle sav
...
         imagesAdapter = adapterFactory.create(items);
         imagesRecyclerView.setAdapter(imagesAdapter);
+        imagesRecyclerView.addOnScrollListener(new RecyclerView.OnScrollListener() {
+            @Override
+            public void onScrollStateChanged(RecyclerView recyclerView, int newState) {
+                super.onScrollStateChanged(recyclerView, newState);
+
+                if (!recyclerView.canScrollVertically(1)) {
+//                    Toast.makeText(ChatDetailsActivity.this,"LAst",Toast.LENGTH_LONG).show();
+                    addImagesToList(query);
```
Comment (neslihanturan) [inline app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java]: Can you clean this comment

**Comment by neslihanturan** [comment]
If you clean comment, I will merge.

File: app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java (hunk 1 above)
Comment (ujjwalagrawal17) [reply app/src/main/java/fr/free/nrw/commons/explore/images/SearchImageFragment.java]: @neslihanturan Done.

**Comment by neslihanturan** [comment]
Thanks @ujjwalagrawal17

**Pull request merged**